lib/src/_shared/hooks/usePickerState.ts

```diff
@@ -81,34 +84,43 @@ export function usePickerState(props: BasePickerProps, options: StateHookOptions
         }
 
         // simulate autoOk, but do not close the modal
-        if (variant === 'inline' || variant === 'static') {
+        if (currentVariant === 'desktop' || currentVariant === 'static') {
           onChange(newDate);
           onAccept && onAccept(newDate);
         }
       },
     }),
-    [acceptDate, autoOk, onAccept, onChange, pickerDate, variant]
+    [acceptDate, autoOk, onAccept, onChange, pickerDate]
   );
 
-  const validationError = validate(value, utils, props);
+  const validationError = validate(value, utils, props as any);
   useEffect(() => {
     if (onError) {
       onError(validationError, value);
     }
   }, [onError, validationError, value]);
 
-  const inputValue = getDisplayDate(date, format, utils, value === null, props);
+  const inputValue = getDisplayDate(value, format, utils, props);
   const inputProps = useMemo(
     () => ({
       inputValue,
+      onChange,
+      format,
+      rawValue: value,
       validationError,
       openPicker: () => !readOnly && !disabled && setIsOpen(true),
     }),
-    [disabled, inputValue, readOnly, setIsOpen, validationError]
+    [disabled, format, inputValue, onChange, readOnly, setIsOpen, validationError, value]
   );
 
   const pickerState = { pickerProps, inputProps, wrapperProps };
+  useDebugValue(pickerState, () => ({
```

Comment:
Interesting, I have almost never used it. I'm wondering, in the dev tools, don't they already scope the state under the name of the hook, do we need the `MuiPickerState` key?